### ISSUE TITLE
No way to import (Anki 2.1.32)

### ISSUE THREAD

**Issue by ran88dom99** [Stvad/CrowdAnki#95]
I tried to import git deck as in example. used https://github.com/Stvad/Software_Engineering__git in CrowdAnki import from github
    Import settings is pretty much empty. No message, no tags, both deckparts at 0. 
	8 personal field sunchecked. two "Basic" headers. first with -61fb1
	I click ok.
	
    Anki 2.1.32 (dee7d45d) Python 3.8.1 Qt 5.12.6 PyQt 5.12.3
    Platform: Windows 8.1
    Flags: frz=True ao=True sv=1
    Add-ons, last update check: 2020-08-28 15:22:56
    
    
    ===Add-ons (active)===
    (add-on provided name [Add-on folder, installed at, version, is config changed])
    Anki Simulator ['817108664', 2020-07-21T10:17, 'None', '']
    CrowdAnki JSON exportimport Edit history Collaborate on deck creation ['1788670778', 2020-07-05T19:26, 'None', '']
    Polar Connect - PDF and Incremental Reading with Anki Sync ['734898866', 2019-01-26T19:01, 'None', '']
    Quantified Self add-on - export your review log fork for Anki 21 ['1806723271', 2020-02-29T08:23, 'None', '']
    True Retention by Card Maturity ['923360400', 2017-08-29T15:00, 'None', '']
    
    ===IDs of active AnkiWeb add-ons===
    1788670778 1806723271 734898866 817108664 923360400
    
    ===Add-ons (inactive)===
    (add-on provided name [Add-on folder, installed at, version, is config changed])
    
Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue on the add-on support site.
Debug info:
Anki 2.1.32 (dee7d45d) Python 3.8.1 Qt 5.12.6 PyQt 5.12.3
Platform: Windows 8.1
Flags: frz=True ao=True sv=1
Add-ons, last update check: 2020-08-28 15:22:56

Caught exception:
Traceback (most recent call last):
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\anki\overrides\change_model_dialog.py", line 169, in accept
    self.collection.models.change(self.old_model, self.note_id_list, self.targetModel, field_map, templates_map)
  File "lib\site-packages\anki\models.py", line 417, in change
AssertionError

now it does not even get that far

Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue on the add-on support site.
Debug info:
Anki 2.1.32 (dee7d45d) Python 3.8.1 Qt 5.12.6 PyQt 5.12.3
Platform: Windows 8.1
Flags: frz=True ao=True sv=1
Add-ons, last update check: 2020-08-28 15:22:56

Caught exception:
Traceback (most recent call last):
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\anki\ui\action_vendor.py", line 35, in <lambda>
    lambda: GitImporter.on_git_import_action(self.window.col))
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\github\github_importer.py", line 28, in on_git_import_action
    GitImporter(collection).import_from_git()
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\github\github_importer.py", line 33, in import_from_git
    self.clone_repository_and_import(repo_url)
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\github\github_importer.py", line 39, in clone_repository_and_import
    porcelain.pull(porcelain.open_repo(str(repo_local_path)), repo_url)
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\dist\dulwich\porcelain.py", line 961, in pull
    fetch_result = client.fetch(
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\dist\dulwich\client.py", line 430, in fetch
    commit()
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\dist\dulwich\client.py", line 416, in commit
    target.object_store.add_thin_pack(f.read, None)
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\dist\dulwich\object_store.py", line 765, in add_thin_pack
    return self._complete_thin_pack(f, path, copier, indexer)
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\dist\dulwich\object_store.py", line 734, in _complete_thin_pack
    index_file.close()
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\dist\dulwich\file.py", line 175, in close
    os.replace(self._lockfilename, self._filename)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\AUSER\\AppData\\Roaming\\Anki2\\addons21\\1788670778\\user_files\\User 1\\Software_Engineering__git\\.git\\objects\\pack\\pack-da39a3ee5e6b4b0d3255bfef95601890afd80709.idx.lock' -> 'C:\\Users\\AUSER\\AppData\\Roaming\\Anki2\\addons21\\1788670778\\user_files\\User 1\\Software_Engineering__git\\.git\\objects\\pack\\pack-da39a3ee5e6b4b0d3255bfef95601890afd80709.idx'

 Tried loading from file it says no such file in directory and errors.
 Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue on the add-on support site.
Debug info:
Anki 2.1.32 (dee7d45d) Python 3.8.1 Qt 5.12.6 PyQt 5.12.3
Platform: Windows 8.1
Flags: frz=True ao=True sv=1
Add-ons, last update check: 2020-08-28 15:22:56

Caught exception:
Traceback (most recent call last):
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\anki\ui\action_vendor.py", line 31, in <lambda>
    lambda: AnkiJsonImporter.import_deck(self.window.col, self.directory_vendor))
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\importer\anki_importer.py", line 112, in import_deck
    AnkiJsonImporter.import_deck_from_path(collection, Path(directory_path))
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\importer\anki_importer.py", line 101, in import_deck_from_path
    if importer.load_deck(directory_path):
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\importer\anki_importer.py", line 29, in load_deck
    deck_json = self.read_deck(self.get_deck_path(directory_path))
  File "C:\Users\AUSER\AppData\Roaming\Anki2\addons21\1788670778\importer\anki_importer.py", line 76, in read_deck
    raise ValueError("There is no {} file inside of the selected directory".format(file_path))
ValueError: There is no D:\Anki3\Software_Engineering__git-master\Software_Engineering__git-master.json file inside of the selected directory

I have a 9 month old install of Anaconda if that is the issue.

**Comment by ran88dom99** [comment]
Software engineering has git sub folder with no cards.

**Comment by ohare93** [comment]
The PermissionError is strange :thinking: though this is probably somewhat related to #92, if only that it is the constant updates to the structure of Anki that are the cause. [As I suggested in that issue](https://github.com/Stvad/CrowdAnki/issues/92#issuecomment-674562491), if at all possible for you, going down to the ~~latest~~~previous stable release 2.1.26 should solve the issue immediately.

I see that this new version 2.1.32 has been labelled the next stable release. @Stvad perhaps fixing these issues to target 32 as the next step is worth some effort :+1: I am quote loaded down with work myself just now, but happy to test fixes if anyone wants to test their changes :+1:

**Comment by MaaxGr** [comment]
```
Debug info:
Anki 2.1.35 (84dcaa86) Python 3.8.0 Qt 5.14.2 PyQt 5.14.2
Platform: Windows 10
Flags: frz=True ao=True sv=1
Add-ons, last update check: 2020-10-05 10:17:05

Caught exception:
Traceback (most recent call last):
  File "C:\Users\max\AppData\Roaming\Anki2\addons21\1788670778\anki\overrides\change_model_dialog.py", line 169, in accept
    self.collection.models.change(self.old_model, self.note_id_list, self.targetModel, field_map, templates_map)
  File "anki\models.py", line 417, in change
AssertionError
```

Seems to be the same Problem for me 2.1.35

**Comment by jukrauss** [comment]
I have the same problem with the same error.
A few days ago the import worked with 2.1.26 for me. Today 2.1.26 is not supported. I also tried 2.25, 2.1.28, 2.1.34 and 2.1.35. No success...

Does anyone have any advice?

**Comment by aplaice** [comment]
Have you tried the latest version of CrowdAnki, (with Anki 2.1.28 or later)? (`Tools` > `Addons` > `Check for Updates`, and then restart Anki.) I've now double checked with a newly installed (as well as an upgraded — not that it should matter) CrowdAnki from AnkiWeb, with Anki 2.1.35, and I can't reproduce the error. (Disclaimer: I wrote the fix, so it is likely to work on my computer, but it shouldn't be machine dependent.) 

@Stvad it seems that you accidentally removed compatibility with 2.1.26, when uploading the newest version, [though](https://ankiweb.net/shared/info/1788670778):

```
Supported Anki versions:

    2.1.0-2.1.25 (updated 2020-06-15)
    2.1.28-2.1.35+ (updated 2020-10-16)
```

**Comment by aplaice** [comment]
Sorry for "bumping", but I've now tested on both Linux and Windows, on different computers, with Anki 2.1.35 (as well several other versions on Linux), and I'm unable to reproduce the error, so if you can reproduce it (with the latest CrowdAnki), please comment here, in order that I can (try to) fix it! (If the error is fixed for you, then ideally also please indicate that! :))

(CrowdAnki on ankiweb is indeed currently incompatible with Anki 2.1.26, so 2.1.26 won't work.)

**Comment by jukrauss** [comment]
Oh sorry. Yes it worked for me at the end with version 2.1.35, as you told me.

**Comment by aplaice** [comment]
> Oh sorry. Yes it worked for me at the end with version 2.1.35, as you told me.

Great! Thanks for replying! :)

**Comment by aplaice** [comment]
There seem to have been several different issues, in this thread.

Some were fixed previously; the git import issue was fixed in #139. I _think_ that there are no further unresolved problems, here, so I'm closing. (Obviously, please complain if I'm wrong.)
